File: quickstart/src/dev.ts
import Nile from "@theniledev/js";

var emoji = require('node-emoji');

import * as dotenv from 'dotenv';

dotenv.config({ override: true })

let envParams = [
  "NILE_URL",
  "NILE_WORKSPACE",
  "NILE_DEVELOPER_EMAIL",
  "NILE_DEVELOPER_PASSWORD",
]
envParams.forEach( (key: string) => {
  if (!process.env[key]) {
    console.error(emoji.get('x'), `Error: missing environment variable ${ key }. See .env.defaults for more info and copy it to .env with your values`);
    process.exit(1);
  }
});

const NILE_URL = process.env.NILE_URL!;
const NILE_WORKSPACE = process.env.NILE_WORKSPACE!;
const NILE_DEVELOPER_EMAIL = process.env.NILE_DEVELOPER_EMAIL!;
const NILE_DEVELOPER_PASSWORD = process.env.NILE_DEVELOPER_PASSWORD!;
const nile!;

// Workflow for the Nile developer
async function createDeveloper() {

  nile = await Nile({
    basePath: NILE_URL,
    workspace: NILE_WORKSPACE,
  });

  // Signup developer
  try {
    await nile.developers.createDeveloper({
      createUserRequest : {
        email : NILE_DEVELOPER_EMAIL,
        password : NILE_DEVELOPER_PASSWORD,
      }
    })
    console.log(emoji.get('white_check_mark'), `Signed up for Nile at ${NILE_URL} as developer ${NILE_DEVELOPER_EMAIL}`);
  } catch (error:any) {
    if (error.message == "user already exists") {
      console.log(emoji.get('dart'), `Developer ${NILE_DEVELOPER_EMAIL} already exists`);
    } else {
      console.error(error);
      process.exit(1);
    }
  };

}

createDeveloper();

File: quickstart-python/src/dev.py
#! /usr/bin/env python

import json
import os
import sys

from dotenv import load_dotenv
from emoji import emojize

from nile_api import AuthenticatedClient, Client
from nile_api.api.developers import create_developer
from nile_api.api.users import create_user, list_users
from nile_api.models.create_user_request import CreateUserRequest

GOOD = emojize(":check_mark_button:")
BAD = emojize(":red_circle:")
DART = emojize(":dart:", language="alias")
ARROW_RIGHT = emojize(":arrow_right:", language="alias")

load_dotenv(override=True)
params = {
    param: os.environ.get(param)
    for param in [
        "NILE_URL",
        "NILE_WORKSPACE",
        "NILE_DEVELOPER_EMAIL",
        "NILE_DEVELOPER_PASSWORD",
    ]
}
NILE_ENTITY_NAME = "DB"

for name, value in params.items():
    if not value:
        print(
            f"{BAD} Error: missing environment variable {name}. See .env.defaults for more info and copy it to .env with your values",  # noqa: E501
        )
        sys.exit(1)


def run():
    # Signup developer
    response = create_developer.sync_detailed(
        client=Client(base_url=params["NILE_URL"]),
        json_body=CreateUserRequest(
            email=params["NILE_DEVELOPER_EMAIL"],
            password=params["NILE_DEVELOPER_PASSWORD"],
        ),
    )
    if response.parsed is not None:
        print(f"{GOOD} Signed up for Nile at {params['NILE_URL']} as developer {params['NILE_DEVELOPER_EMAIL']}")
    else:
        error = json.loads(response.content)
        if error["message"] == "user already exists":
            print(f"{DART} Developer {params['NILE_DEVELOPER_EMAIL']} already exists")
        else:
            print(f"{BAD} Error: {error.message}")
            sys.exit(1)


if __name__ == "__main__":
    run()
